turbopy/core.py

```diff
@@ -724,6 +734,89 @@ def interpval(yvec):
             return interpval
 
 
+    def set_cell_volumes(self, coordinate_system):
+        """
+        This function sets some volume-related elements that are useful for
+        finite difference solvers in curvilinear coordinates.
+
+        This function set the following volume-realted quantities:
+
+        old_cell_volume(cell.size) - These are the volume elements at the beginning
+                                     of the time step.
+        cell_volume(cell.size)     - These are the volume elements at the end
+                                     of the time step.
+        interface_volume(edge.size) - Average of adjacent cell volumes
+        inverse_cell_volume(cell.size)   -  1/cell_volume
+        """
+        old_grid = self.cell_edges  # Old grid at beginning of time step
+        new_grid = self.cell_edges  # New grid at end of time step
+        fourthirds = 4/0/3.0
+        if coordinate_system.lower().strip() == 'cartesian':
```

Comment:
Use `self.coordinate_system` here. If you really think that `lower` and `strip` should be applied, then apply them when you save the parameter during initialization:
```python
self.coordinate_system = input_data['coordinate_system'].lower().strip()
```

Reply:
This is still not resolved.